### PR TITLE
Added [Imports] flake8-internal-name-import

### PR DESCRIPTION
**Pull request by rows-s**
Hope some would find [the plugin](https://github.com/rows-s/flake8_private_name_import) helpful.

Plugin reports imports of internal names, modules and imports from internal modules. Where "internal" is a name or module that starts with "_" but is not dunder (`__future__`, `__all__`, etc.).

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Suggestion: in Python we use "private" for `__private` names, it is better to use "protected" in this case for `_protected` names.

**Comment by rows-s** [comment]
@sobolevn
> Suggestion: in Python we use "private" for `__private` names, it is better to use "protected" in this case for `_protected` names.

As i know, "private" and "protected" applies only for classes' attributes but not for names (variables, functions, classes).
In python if class has a `__private` attribute than at runtime the attribute would be prefixed with `_{class.__name__}` as well as code that accesses them.
But those attributes are not really "private", they are still accessible with the prefix. And it only help to fix collision with names used in sub-classes (only if classes named differently).

For module-level names there is not difference between `_protected` and `__private`. You can import them as is without any prefix. And so i think there is not need to call them differently. 

But i don mind to have the description as `Plugin that reports imports of protected and private names.`

# Example of private attributes  

```python
>>> class A:
...    _protected: str = 'protected'
...    __private: str = 'private'
...    
...    def get_private(self) -> str:
...        return self.__private
...   
... class B(A):
...    _protected: str = 'new protected'
...    __private: str = 'new private'
...    
...    def get_new_private(self) -> str:
...        return self.__private

>>> b = B()
>>> b._protected, b.get_private(), b.get_new_private()
('new protected', 'private', 'new private')
>>>b._protected, b._A__private, b._B__private
('new protected', 'private', 'new private')
```

**Comment by orsinium** [comment]
Thank you for your contribution! According to our [contribution guidelines](https://github.com/DmytroLitvinov/awesome-flake8-extensions/blob/master/contributing.md), the package must be at least 30 days old before it can be added to the list. Let it boild for a while :)

**Comment by orsinium** [comment]
> In Python we use "private" for `__private` names, it is better to use "protected" in this case for `_protected` names.

I call them "underscored" and "double underscored" 🌚 IDK if anyone else ever does that but people usually understand what I mean. 

If you want to 100% avoid confusion, you can call it "underscore-prefixed" and "double-underscore-prefixed", which is a mouthful but gets the idea across even for people who don't know the fancy terms.

**Comment by rows-s** [comment]
Probably it was better to call it "non-public names" or "internal names".

My fault, should check [python docs](https://docs.python.org/3/tutorial/classes.html#private-variables) before.

Maybe I will rename within that 30 days :)

**Comment by rows-s** [comment]
Hello! 👋

Repository lives for more than a month already. 
Won't you mind to merge this PR?

**Review by orsinium — APPROVED** [review]
Looks good to me, thank you :)

File: README.md [hunk 1]
```diff
@@ -174,6 +174,7 @@ Extensions for checking import statements.
 - [flake8-import-style](https://github.com/sfstpala/flake8-import-style) - Plugin to ensure explicit module imports.
 - [flake8-tidy-imports](https://github.com/adamchainz/flake8-tidy-imports) - Extension that helps you write tidier imports.
 - [flake8-type-checking](https://github.com/sondrelg/flake8-type-checking) - Plugin for managing type-checking imports & forward references.
+- [flake8-internal-name-import](https://github.com/rows-s/flake8_internal_name_import) - Plugin that reports imports of internal names.
```
Comment (sobolevn) [inline README.md:177]:
```suggestion
- [flake8-internal-name-import](https://github.com/rows-s/flake8_internal_name_import) - Plugin that reports imports of protected names.
```

**Pull request merged**